### PR TITLE
Visibilty on UI_ToggleGamePanels fixes

### PR DESCRIPTION
**Pull request by Ejsstiil**
Visibilty on UI_ToggleGamePanels fixes:
- properly hides the panel - namely the child icons and the open arrow(when panel is toggled off and ui is hidden)
- improved behavior of the initial state based on the last saved state
 - when the last state was hidden: no longer draws thin side of the panel on new game
 - when the last state was hidden: no longer needed to double press hotkey to show it again
Tested on empty profile, with no mods

image

### CODE REVIEW

**Comment by Garanas** [comment]
@4z0t can you review and merge accordingly?

File: lua/ui/game/score.lua [hunk 1]
```diff
@@ -765,11 +743,30 @@ function ToggleScoreControl(state)
...
+                else
+                    self.Right:Set(newRight)
+                end
+            end
         else
             Prefs.SetToCurrentProfile("scoreoverlay", false)
             local sound = Sound({Cue = "UI_Score_Window_Close", Bank = "Interface",})
             PlaySound(sound)
             controls.bg:SetNeedsFrameUpdate(true)
+            controls.bg.OnFrame = function(self, delta)
```
Comment (Garanas) [inline lua/ui/game/score.lua:760]: Could you elaborate why this change was required?

Comment (Ejsstiil) [reply lua/ui/game/score.lua:760]: This fixes restoration if the last-state (collapsed/expanded)
there were multiple issues:
- when you exited game with hidden SB, on the next start of game:
   - small portion of bar is visible upon new start image
   - in this state when you press F2, it wants to collapse first, despite its already "collapsed" 
   - arrow stayed visible on UI_ToggleGamePanels when state is Collapsed

https://user-images.githubusercontent.com/36369441/171967983-d932de9c-62c7-436f-9c81-eeb40b2e27f3.mp4
